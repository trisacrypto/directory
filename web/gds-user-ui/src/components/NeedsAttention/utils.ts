export const enum AttentionAction {
  'VERIFY_EMAILS' = 'VERIFY_EMAILS',
  'VERIFY_PHONES' = 'VERIFY_PHONES',
  'VERIFY_ADDRESSES' = 'VERIFY_ADDRESSES',
  'VERIFY_IDENTITIES' = 'VERIFY_IDENTITIES',
  'VERIFY_DOCUMENTS' = 'VERIFY_DOCUMENTS',
  'VERIFY_ALL' = 'VERIFY_ALL',
  'VERIFY_NONE' = 'VERIFY_NONE'
}
