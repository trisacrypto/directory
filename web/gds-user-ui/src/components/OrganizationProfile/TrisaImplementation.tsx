import React, { useState } from 'react';
import { Heading, Stack, Table, Tbody, Tr, Td, Text } from '@chakra-ui/react';
import { Trans } from '@lingui/react';
type TrisaImplementationProps = {
  data: any;
};
const TrisaImplementation: React.FC<TrisaImplementationProps> = ({ data }) => {
  return (
    <Stack
      border="1px solid #DFE0EB"
      fontFamily={'Open Sans'}
      bg={'white'}
      color={'#252733'}
      fontSize={18}
      p={4}
      mb={7}
      px={7}>
      <Stack width={'100%'}>
        <Heading as={'h1'} fontSize={19} pb={2} pt={4}>
          {' '}
          TRISA Implementations{' '}
        </Heading>
        <Text pb={7}>
          You must request a new X.509 Identity Certificate to change your Enpoint and Common Name.
        </Text>
        <Stack fontSize={18}>
          <Table
            sx={{
              'td:nth-child(1)': { fontWeight: 'bold' },
              Tr: {
                borderStyle: 'hidden'
              }
            }}>
            <Tbody
              sx={{
                '*': {
                  fontSize: '1rem',
                  'td:first-child': {
                    width: '50%'
                  },
                  td: {
                    borderBottom: 'none',
                    paddingInlineStart: 0,
                    paddingY: 2.5
                  }
                }
              }}>
              <Tr pt={'1rem !important'}>
                <Td colSpan={2} background="#E5EDF1" fontWeight="bold" pl={'1rem !important'}>
                  <Trans id="TestNet">TestNet</Trans>
                </Td>
              </Tr>
              <Tr>
                <Td pt={'1rem !important'} pl={'1rem !important'}>
                  <Trans id="Endpoint">Endpoint</Trans>
                </Td>
                <Td>{data?.testnet?.endpoint || 'N/A'}</Td>
              </Tr>
              <Tr>
                <Td pl={'1rem !important'}>
                  <Trans id="Common Name">Common Name</Trans>
                </Td>
                <Td>{data?.testnet?.common_name || 'N/A'}</Td>
              </Tr>
              <Tr>
                <Td colSpan={2}></Td>
              </Tr>
              <Tr>
                <Td colSpan={2} background="#E5EDF1" fontWeight="bold" pl={'1rem !important'}>
                  <Trans id="MainNet">MainNet</Trans>
                </Td>
              </Tr>
              <Tr>
                <Td pt={'1rem !important'} pl={'1rem !important'}>
                  <Trans id="Endpoint">Endpoint</Trans>
                  Endpoint
                </Td>
                <Td>{data?.mainnet?.endpoint || 'N/A'}</Td>
              </Tr>
              <Tr>
                <Td pl={'1rem !important'}>
                  <Trans id="Common Name">Common Name</Trans>
                </Td>
                <Td>{data?.mainnet?.common_name || 'N/A'}</Td>
              </Tr>
            </Tbody>
          </Table>
        </Stack>
      </Stack>
    </Stack>
  );
};
export default TrisaImplementation;
