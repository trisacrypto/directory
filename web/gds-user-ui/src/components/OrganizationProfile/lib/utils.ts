export const StatusCheck = (status: string) => {
  switch (status) {
    case 'NO_VERIFICATION':
      return 'Not Verified';
    case 'PENDING_REVIEW':
      return 'Pending Review';
    case 'VERIFIED':
      return 'Verified';
    case 'REJECTED':
      return 'Rejected';
    default:
      return 'Pending Registration';
  }
};
