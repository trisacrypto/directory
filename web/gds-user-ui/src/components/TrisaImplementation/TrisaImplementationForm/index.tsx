import { WarningIcon } from '@chakra-ui/icons';
import { Heading, Stack, Text } from '@chakra-ui/react';
import InputFormControl from 'components/ui/InputFormControl';
import React from 'react';
import { useFormContext } from 'react-hook-form';
import { getDomain } from 'utils/utils';
import { t } from '@lingui/macro';

type TrisaImplementationFormProps = {
  headerText: string;
  name: string;
  type: 'TestNet' | 'MainNet';
};

const env = {
  TestNet: 'testnet',
  MainNet: 'trisa'
};

const TrisaImplementationForm: React.FC<TrisaImplementationFormProps> = ({
  headerText,
  name,
  type
}) => {
  const {
    register,
    formState: { errors },
    watch,
    getValues
  } = useFormContext();
  const commonName = watch(`${name}.common_name`);
  const trisaEndpoint = watch(`${name}.endpoint`);
  const [commonNameWarning, setCommonNameWarning] = React.useState<string | undefined>('');

  React.useEffect(() => {
    const trisaEndpointUri = trisaEndpoint?.split(':')[0];

    const warningMessage =
      trisaEndpointUri === commonName
        ? undefined
        : t`Common name should match the TRISA endpoint without the port.`;
    setCommonNameWarning(warningMessage);
  }, [commonName, trisaEndpoint]);

  const getCommonNameFormHelperText = () => {
    if ((errors[name] as any)?.common_name) {
      return (errors[name] as any)?.common_name.message;
    }
    if (commonNameWarning) {
      return (
        <Text color="yellow.500">
          <WarningIcon /> {commonNameWarning}
        </Text>
      );
    }

    return t`The common name for the mTLS certificate. This should match the TRISA endpoint without the port in most cases.`;
  };

  const domain = getValues('website') && getDomain(getValues('website'));

  return (
    <Stack>
      <Heading size="md">{headerText}</Heading>
      <InputFormControl
        label={t`TRISA Endpoint`}
        placeholder={`${env[type]}.${domain}:443`}
        formHelperText={
          (errors[name] as any)?.endpoint
            ? (errors[name] as any)?.endpoint?.message
            : t`The address and port of the TRISA endpoint for partner VASPs to connect on via gRPC.`
        }
        controlId="trisaEndpoint"
        isInvalid={!!(errors[name] as any)?.endpoint}
        {...register(`${name}.endpoint`)}
      />

      <InputFormControl
        label={t`Certificate Common Name`}
        placeholder={`${env[type]}.${domain}`}
        isInvalid={!!(errors[name] as any)?.common_name}
        formHelperText={getCommonNameFormHelperText()}
        controlId="certificateCommonName"
        {...register(`${name}.common_name`)}
      />
    </Stack>
  );
};

export default TrisaImplementationForm;
