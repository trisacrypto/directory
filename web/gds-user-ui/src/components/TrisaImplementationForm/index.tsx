import { WarningIcon } from '@chakra-ui/icons';
import { Heading, Text } from '@chakra-ui/react';
import InputFormControl from 'components/ui/InputFormControl';
import FormLayout from 'layouts/FormLayout';
import React from 'react';
import { useFormContext } from 'react-hook-form';
import { getDomain } from 'utils/utils';
import _ from 'lodash/fp';

type TrisaImplementationFormProps = {
  headerText: string;
  name: string;
  type: 'TestNet' | 'MainNet';
};

const env = {
  TestNet: 'testnet',
  MainNet: 'trisa'
};

const TrisaImplementationForm: React.FC<TrisaImplementationFormProps> = ({
  headerText,
  name,
  type
}) => {
  const {
    register,
    formState: { errors },
    watch,
    setError,
    getValues
  } = useFormContext();
  const commonName = watch(`${name}.common_name`);
  const trisaEndpoint = watch(`${name}.endpoint`);
  const [commonNameWarning, setCommonNameWarning] = React.useState<string | undefined>('');
  React.useEffect(() => {
    const trisaEndpointUri = trisaEndpoint?.split(':')[0];

    const warningMessage =
      trisaEndpointUri === commonName
        ? undefined
        : 'common name should match the TRISA endpoint without the port';
    setCommonNameWarning(warningMessage);
  }, [commonName, trisaEndpoint]);

  const getCommonNameFormHelperText = () => {
    if (errors[name]?.common_name) {
      return errors[name]?.common_name.message;
    }
    if (commonNameWarning) {
      return (
        <Text color="yellow.500">
          <WarningIcon /> {commonNameWarning}
        </Text>
      );
    }

    return 'The common name for the mTLS certificate. This should match the TRISA endpoint without the port in most cases.';
  };

  const domain = getValues('website') && getDomain(getValues('website'));

  return (
    <FormLayout>
      <Heading size="md">{headerText}</Heading>
      <InputFormControl
        label="TRISA Endpoint"
        placeholder={`${env[type]}.${domain}:443`}
        formHelperText={
          errors[name]?.endpoint
            ? errors[name]?.endpoint?.message
            : 'The address and port of the TRISA endpoint for partner VASPs to connect on via gRPC.'
        }
        controlId="trisaEndpoint"
        isInvalid={!!errors[name]?.endpoint}
        {...register(`${name}.endpoint`)}
      />

      <InputFormControl
        label="Certificate Common Name"
        placeholder={`${env[type]}.${domain}`}
        isInvalid={!!errors[name]?.common_name}
        formHelperText={getCommonNameFormHelperText()}
        controlId="certificateCommonName"
        {...register(`${name}.common_name`)}
      />
    </FormLayout>
  );
};

export default TrisaImplementationForm;
