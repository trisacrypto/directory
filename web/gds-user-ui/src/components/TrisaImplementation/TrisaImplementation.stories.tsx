import { Meta, Story } from "@storybook/react";
import TrisaImplementation from ".";

export default {
  title: "components/Trisa Implementation",
  component: TrisaImplementation,
} as Meta;

const Template: Story = (args) => <TrisaImplementation {...args} />;

export const Default = Template.bind({});
Default.args = {};
