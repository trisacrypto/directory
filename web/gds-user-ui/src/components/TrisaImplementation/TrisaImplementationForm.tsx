import React, { useEffect, useState, useRef, Dispatch, SetStateAction } from 'react';
import { chakra, useDisclosure } from '@chakra-ui/react';
import { t } from '@lingui/macro';
import FormLayout from 'layouts/FormLayout';
import { FormProvider, useForm } from 'react-hook-form';
import { trisaImplementationValidationSchema } from 'modules/dashboard/certificate/lib/trisaImplementationValidationSchema';
import { yupResolver } from '@hookform/resolvers/yup';
import useCertificateStepper from 'hooks/useCertificateStepper';
import StepButtons from 'components/StepsButtons';
import { StepEnum } from 'types/enums';
import { useUpdateCertificateStep } from 'hooks/useUpdateCertificateStep';
import TrisaImplementationForm from './TrisaImplementationForm/index';
import { StepsIndexes } from 'constants/steps';
import { isProdEnv } from 'application/config';
import { DevTool } from '@hookform/devtools';
interface TrisaFormProps {
  data?: any;
  isLoading?: boolean;
  shouldResetForm?: boolean;
  onResetFormState?: Dispatch<SetStateAction<boolean>>;
}
const TrisaForm: React.FC<TrisaFormProps> = ({ data, shouldResetForm, onResetFormState }) => {
  const { isOpen, onClose, onOpen } = useDisclosure();
  const [shouldShowResetFormModal, setShouldShowResetFormModal] = useState(false);
  const { previousStep, nextStep, currentState, updateIsDirty } = useCertificateStepper();

  const {
    updateCertificateStep,
    updatedCertificateStep,
    wasCertificateStepUpdated,
    isUpdatingCertificateStep,
    reset: resetMutation
  } = useUpdateCertificateStep();
  const previousStepRef = useRef<any>(false);
  const nextStepRef = useRef<any>(false);
  const resolver = yupResolver(trisaImplementationValidationSchema);
  const methods = useForm({
    defaultValues: data,
    resolver,
    mode: 'onChange'
  });

  const {
    formState: { isDirty },
    reset: resetForm
  } = methods;

  useEffect(() => {
    updateIsDirty(isDirty, StepsIndexes.TRISA_IMPLEMENTATION);
  }, [isDirty, updateIsDirty]);

  useEffect(() => {
    if (shouldShowResetFormModal) {
      onOpen();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [shouldShowResetFormModal]);

  if (wasCertificateStepUpdated && nextStepRef.current) {
    resetMutation();
    // reset the form with the new values
    resetForm(updatedCertificateStep?.form, {
      keepValues: false
    });
    nextStep(updatedCertificateStep);
    nextStepRef.current = false;
  }

  if (wasCertificateStepUpdated && previousStepRef.current && !isUpdatingCertificateStep) {
    resetMutation();
    // reset the form with the new values
    resetForm(updatedCertificateStep?.form);
    console.log('[] prev updatedCertificateStep', updatedCertificateStep);
    previousStepRef.current = false;
    previousStep(updatedCertificateStep);
  }

  const handlePreviousStepClick = () => {
    if (isDirty) {
      const payload = {
        step: StepEnum.TRISA,
        form: {
          ...methods.getValues(),
          state: currentState()
        } as any
      };
      updateCertificateStep(payload);
      previousStepRef.current = true;
    }
    previousStep(data);
  };

  const handleNextStepClick = () => {
    if (!isDirty) {
      nextStep(data);
    } else {
      const payload = {
        step: StepEnum.TRISA,
        form: {
          ...methods.getValues(),
          state: currentState()
        } as any
      };

      updateCertificateStep(payload);
      nextStepRef.current = true;
    }
  };
  const handleResetForm = () => {
    setShouldShowResetFormModal(true); // this will show the modal
  };

  const handleResetClick = () => {
    setShouldShowResetFormModal(false); // this will close the modal
  };
  const onCloseModalHandler = () => {
    setShouldShowResetFormModal(false);
    onClose();
  };

  // reset the form from the parent component
  useEffect(() => {
    if (shouldResetForm && onResetFormState) {
      resetForm(data);
      onResetFormState(false);
    }
  }, [shouldResetForm, resetForm, data, onResetFormState]);

  return (
    <FormLayout>
      <FormProvider {...methods}>
        <chakra.form
          onSubmit={methods.handleSubmit(handleNextStepClick)}
          data-testid="trisa-implementation-form">
          <TrisaImplementationForm
            type="TestNet"
            name="testnet"
            headerText={t`TRISA Endpoint: TestNet`}
          />
          <TrisaImplementationForm
            type="MainNet"
            name="mainnet"
            headerText={t`TRISA Endpoint: MainNet`}
          />
          <StepButtons
            handlePreviousStep={handlePreviousStepClick}
            handleNextStep={handleNextStepClick}
            onResetModalClose={handleResetClick}
            isOpened={isOpen}
            handleResetForm={handleResetForm}
            resetFormType={StepEnum.TRISA}
            onClosed={onCloseModalHandler}
            handleResetClick={handleResetClick}
            shouldShowResetFormModal={shouldShowResetFormModal}
          />
        </chakra.form>
        {!isProdEnv ? <DevTool control={methods.control} /> : null}
      </FormProvider>
    </FormLayout>
  );
};

export default TrisaForm;
