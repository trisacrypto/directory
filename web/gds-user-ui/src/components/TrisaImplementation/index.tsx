import { InfoIcon } from '@chakra-ui/icons';
import { Box, Heading, HStack, Icon, Stack, Text } from '@chakra-ui/react';
import { getSteps, getCurrentStep } from 'application/store/selectors/stepper';
import { SectionStatus } from 'components/SectionStatus';
import TrisaImplementationForm from 'components/TrisaImplementationForm';
import FormLayout from 'layouts/FormLayout';
import { useSelector } from 'react-redux';
import { getStepStatus } from 'utils/utils';

const TrisaImplementation: React.FC = () => {
  const steps = useSelector(getSteps);
  const currentStep = useSelector(getCurrentStep);
  const stepStatus = getStepStatus(steps, currentStep);

  return (
    <Stack spacing={7} pt={8}>
      <HStack>
        <Heading size="md">Section 4: TRISA Implementation</Heading>
        {stepStatus ? <SectionStatus status={stepStatus} /> : null}
      </HStack>
      <FormLayout>
        <Text>
          Each VASP is required to establish a TRISA endpoint for inter-VASP communication. Please
          specify the details of your endpoint for certificate issuance. Please specify the TestNet
          endpoint and the MainNet endpoint. The TestNet endpoint and the MainNet endpoint must be
          different.
        </Text>
      </FormLayout>
      <TrisaImplementationForm
        type="TestNet"
        name="trisa_endpoint_testnet"
        headerText="TRISA Endpoint: TestNet"
      />
      <TrisaImplementationForm
        type="MainNet"
        name="trisa_endpoint_mainnet"
        headerText="TRISA Endpoint: MainNet"
      />
    </Stack>
  );
};

export default TrisaImplementation;
