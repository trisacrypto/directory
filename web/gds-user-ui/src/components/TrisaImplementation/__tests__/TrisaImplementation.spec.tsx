import { act, getByRole, render, screen } from "utils/test-utils";
import TrisaImplementationForm from "../TrisaImplementationForm";
import { dynamicActivate } from "utils/i18nLoaderHelper";
import TrisaImplementation from "..";

function renderComponent() {
    return render(<TrisaImplementationForm />);
}

jest.mock("hooks/useFetchCertificateStep", () => ({
    useFetchCertificateStep: () => ({
        certificateStep: {
            form: jest.fn(),
            errors: jest.fn(),
        },
        isFetchingCertificateStep: false,
    }),
}));

describe("TrisaImplementation", () => {
    beforeAll(() => {
        act(() => {
            dynamicActivate("en");
        });
    });

    it("should render", () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
    });

    it("should render the form", () => {
        const { getByTestId } = renderComponent();
        const trisaImplementation = getByTestId("trisa-implementation-form");
        expect(trisaImplementation).toBeInTheDocument();
    });

    it("should render the TestNet header", () => {
        const { getByText } = renderComponent();
        const testNet = getByText("TRISA Endpoint: TestNet");
        expect(testNet).toBeInTheDocument();
    });

    it("should render the MainNet header", () => {
        const { getByText } = renderComponent();
        const mainNet = getByText("TRISA Endpoint: MainNet");
        expect(mainNet).toBeInTheDocument();
    });

    it("should render StepButtons", () => {
        const { getByTestId } = renderComponent();
        const stepButtons = getByTestId("step-buttons");
        expect(stepButtons).toBeInTheDocument();
    });
});