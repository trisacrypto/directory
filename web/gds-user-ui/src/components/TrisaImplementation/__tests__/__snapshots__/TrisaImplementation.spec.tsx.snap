// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrisaImplementation should render 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: var(--chakra-colors-white);
  border: 2px solid #E5EDF1;
  border-radius: 10px;
  padding: var(--chakra-space-3);
}

.emotion-0>*:not(style)~*:not(style) {
  margin-top: var(--chakra-space-3-5);
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

@media screen and (min-width: 48em) {
  .emotion-0 {
    padding: var(--chakra-space-9);
  }
}

.emotion-3 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: var(--chakra-fontSizes-xl);
  line-height: 1.2;
}

.emotion-4 {
  width: 100%;
  position: relative;
}

.emotion-5 {
  display: block;
  text-align: start;
  font-size: var(--chakra-fontSizes-md);
  -webkit-margin-end: var(--chakra-space-3);
  margin-inline-end: var(--chakra-space-3);
  margin-bottom: var(--chakra-space-2);
  font-weight: var(--chakra-fontWeights-medium);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  opacity: 1;
}

.emotion-5[disabled],
.emotion-5[aria-disabled=true],
.emotion-5[data-disabled] {
  opacity: 0.4;
}

.emotion-6 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.emotion-7 {
  width: 100%;
  min-width: 0px;
  outline: 2px solid transparent;
  outline-offset: 2px;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  height: var(--chakra-sizes-10);
  border-radius: 0px;
  border: 1px solid;
  border-color: inherit;
  background: #E3EBEF;
}

.emotion-7:hover,
.emotion-7[data-hover] {
  border-color: var(--chakra-colors-gray-300);
}

.emotion-7[aria-readonly=true],
.emotion-7[readonly],
.emotion-7[data-readonly] {
  box-shadow: none!important;
  -webkit-user-select: all;
  -moz-user-select: all;
  -ms-user-select: all;
  user-select: all;
}

.emotion-7[disabled],
.emotion-7[aria-disabled=true],
.emotion-7[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
}

.emotion-7[aria-invalid=true],
.emotion-7[data-invalid] {
  border-color: #E53E3E;
  box-shadow: 0 0 0 1px #E53E3E;
}

.emotion-7:focus,
.emotion-7[data-focus] {
  z-index: 1;
  border-color: #3182ce;
  box-shadow: 0 0 0 1px #3182ce;
}

.emotion-8 {
  margin-top: var(--chakra-space-2);
  color: var(--chakra-colors-gray-500);
  line-height: var(--chakra-lineHeights-normal);
  font-size: var(--chakra-fontSizes-sm);
}

.emotion-26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  padding-top: var(--chakra-space-6);
  padding-bottom: var(--chakra-space-6);
  row-gap: var(--chakra-space-2);
}

.emotion-26>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: var(--chakra-space-8);
  margin-inline-start: var(--chakra-space-8);
}

.emotion-27 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-gray-100);
}

.emotion-27:focus,
.emotion-27[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-27[disabled],
.emotion-27[aria-disabled=true],
.emotion-27[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-27:hover,
.emotion-27[data-hover] {
  background: var(--chakra-colors-gray-200);
}

.emotion-27:hover[disabled],
.emotion-27[data-hover][disabled],
.emotion-27:hover[aria-disabled=true],
.emotion-27[data-hover][aria-disabled=true],
.emotion-27:hover[data-disabled],
.emotion-27[data-hover][data-disabled] {
  background: var(--chakra-colors-gray-100);
}

.emotion-27:active,
.emotion-27[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-28 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
}

.emotion-28:focus,
.emotion-28[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-28[disabled],
.emotion-28[aria-disabled=true],
.emotion-28[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-28:hover[disabled],
.emotion-28[data-hover][disabled],
.emotion-28:hover[aria-disabled=true],
.emotion-28[data-hover][aria-disabled=true],
.emotion-28:hover[data-disabled],
.emotion-28[data-hover][data-disabled] {
  background: initial;
}

<div>
  <div
    class="chakra-stack emotion-0"
  >
    <form
      class="emotion-1"
      data-testid="trisa-implementation-form"
    >
      <div
        class="chakra-stack emotion-0"
      >
        <h2
          class="chakra-heading emotion-3"
        >
          TRISA Endpoint: TestNet
        </h2>
        <div
          class="chakra-form-control emotion-4"
          role="group"
        >
          <label
            class="chakra-form__label emotion-5"
            for="trisaEndpoint"
            id="field-1-label"
          >
            TRISA Endpoint
          </label>
          <div
            class="chakra-input__group emotion-6"
          >
            <input
              aria-describedby="field-1-helptext"
              class="chakra-input emotion-7"
              id="trisaEndpoint"
              name="testnet.endpoint"
              placeholder="testnet.undefined:443"
              type="text"
            />
          </div>
          <div
            class="chakra-form__helper-text emotion-8"
            id="field-1-helptext"
          >
            The address and port of the TRISA endpoint for partner VASPs to connect on via gRPC.
          </div>
        </div>
        <div
          class="chakra-form-control emotion-4"
          role="group"
        >
          <label
            class="chakra-form__label emotion-5"
            for="certificateCommonName"
            id="field-2-label"
          >
            Certificate Common Name
          </label>
          <div
            class="chakra-input__group emotion-6"
          >
            <input
              aria-describedby="field-2-helptext"
              class="chakra-input emotion-7"
              id="certificateCommonName"
              name="testnet.common_name"
              placeholder="testnet.undefined"
              type="text"
            />
          </div>
          <div
            class="chakra-form__helper-text emotion-8"
            id="field-2-helptext"
          >
            The common name for the mTLS certificate. This should match the TRISA endpoint without the port in most cases.
          </div>
        </div>
      </div>
      <div
        class="chakra-stack emotion-0"
      >
        <h2
          class="chakra-heading emotion-3"
        >
          TRISA Endpoint: MainNet
        </h2>
        <div
          class="chakra-form-control emotion-4"
          role="group"
        >
          <label
            class="chakra-form__label emotion-5"
            for="trisaEndpoint"
            id="field-3-label"
          >
            TRISA Endpoint
          </label>
          <div
            class="chakra-input__group emotion-6"
          >
            <input
              aria-describedby="field-3-helptext"
              class="chakra-input emotion-7"
              id="trisaEndpoint"
              name="mainnet.endpoint"
              placeholder="trisa.undefined:443"
              type="text"
            />
          </div>
          <div
            class="chakra-form__helper-text emotion-8"
            id="field-3-helptext"
          >
            The address and port of the TRISA endpoint for partner VASPs to connect on via gRPC.
          </div>
        </div>
        <div
          class="chakra-form-control emotion-4"
          role="group"
        >
          <label
            class="chakra-form__label emotion-5"
            for="certificateCommonName"
            id="field-4-label"
          >
            Certificate Common Name
          </label>
          <div
            class="chakra-input__group emotion-6"
          >
            <input
              aria-describedby="field-4-helptext"
              class="chakra-input emotion-7"
              id="certificateCommonName"
              name="mainnet.common_name"
              placeholder="trisa.undefined"
              type="text"
            />
          </div>
          <div
            class="chakra-form__helper-text emotion-8"
            id="field-4-helptext"
          >
            The common name for the mTLS certificate. This should match the TRISA endpoint without the port in most cases.
          </div>
        </div>
      </div>
      <div
        class="chakra-stack emotion-26"
        data-testid="step-buttons"
      >
        <button
          class="chakra-button emotion-27"
          type="button"
        >
          Save & Previous
        </button>
        <button
          class="chakra-button emotion-28"
          type="button"
        >
          Save & Next
        </button>
        <button
          class="chakra-button emotion-27"
          type="button"
        >
          Clear & Reset Form
        </button>
      </div>
    </form>
  </div>
</div>
`;
